hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/tables/datanodesTable.tsx

```diff
@@ -0,0 +1,315 @@
+        used={record.storageUsed}
+        remaining={record.storageRemaining}
+        committed={record.storageCommitted} />
+    )
+  },
+  {
+    title: 'Last Heartbeat',
+    dataIndex: 'lastHeartbeat',
+    key: 'lastHeartbeat',
+    sorter: (a: Datanode, b: Datanode) => moment(a.lastHeartbeat).unix() - moment(b.lastHeartbeat).unix(),
+    render: (heartbeat: number) => {
+      return heartbeat > 0 ? getTimeDiffFromTimestamp(heartbeat) : 'NA';
+    }
+  },
+  {
+    title: 'Pipeline ID(s)',
+    dataIndex: 'pipelines',
+    key: 'pipelines',
+    render: (pipelines: Pipeline[], record: Datanode) => {
+      const renderPipelineIds = (pipelineIds: Pipeline[]) => {
+        return pipelineIds?.map((pipeline: any, index: any) => (
+          <div key={index} className='pipeline-container-v2'>
+            <ReplicationIcon
+              replicationFactor={pipeline.replicationFactor}
+              replicationType={pipeline.replicationType}
+              leaderNode={pipeline.leaderNode}
+              isLeader={pipeline.leaderNode === record.hostname} />
+            {pipeline.pipelineID}
+          </div >
+        ))
+      }
+
+      return (
+        <Popover
+          content={
+            renderPipelineIds(pipelines)
+          }
+          title="Related Pipelines"
+          placement="bottomLeft"
+          trigger="hover">
+          <strong>{pipelines.length}</strong> pipelines
+        </Popover>
+      );
+    }
+  },
+  {
+    title: () => (
+      <span style={headerIconStyles} >
+        Leader Count
+        <Tooltip
+          title='The number of Ratis Pipelines in which the given datanode is elected as a leader.' >
+          <InfoCircleOutlined style={{ paddingLeft: '4px' }} />
+        </Tooltip>
+      </span>
+    ),
+    dataIndex: 'leaderCount',
+    key: 'leaderCount',
+    sorter: (a: Datanode, b: Datanode) => a.leaderCount - b.leaderCount
+  },
+  {
+    title: 'Containers',
+    dataIndex: 'containers',
+    key: 'containers',
+    sorter: (a: Datanode, b: Datanode) => a.containers - b.containers
+  },
+  {
+    title: () => (
+      <span style={headerIconStyles}>
+        Open Container
+        <Tooltip title='The number of open containers per pipeline.'>
+          <InfoCircleOutlined style={{ paddingLeft: '4px' }} />
+        </Tooltip>
+      </span>
+    ),
+    dataIndex: 'openContainers',
+    key: 'openContainers',
+    sorter: (a: Datanode, b: Datanode) => a.openContainers - b.openContainers
+  },
+  {
+    title: 'Version',
+    dataIndex: 'version',
+    key: 'version',
+    sorter: (a: Datanode, b: Datanode) => a.version.localeCompare(b.version),
+    defaultSortOrder: 'ascend' as const
+  },
+  {
+    title: 'Setup Time',
+    dataIndex: 'setupTime',
+    key: 'setupTime',
+    sorter: (a: Datanode, b: Datanode) => a.setupTime - b.setupTime,
+    render: (uptime: number) => {
+      return uptime > 0 ? moment(uptime).format('ll LTS') : 'NA';
+    }
+  },
+  {
+    title: 'Revision',
```

Comment:
I’m not sure if the searchable Revision field is useful in its current form.

<img width="1139" alt="image" src="https://github.com/user-attachments/assets/0dbb2c01-7e18-4f14-a4b6-588a3e743f14">